### PR TITLE
Assemble and Pack task testing

### PR DESCRIPTION
**Pull request by brunobowden**
- AssembleSourceTask unit tests
- AssembleLibrariesTask unit tests
- PackLibrariesTask unit tests
- MockProjectExec added demandCopyAndReturn(CopySpec closure)

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - PTAL. Another round of testing.

**Comment by brunobowden** [comment]
We now have coverage for all the tasks. The only things missing unit test coverage are NativeCompilation and J2objcPlugin. Both of which are more difficult to test.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTask.groovy [hunk 1]
```diff
@@ -54,23 +54,19 @@ class PackLibrariesTask extends DefaultTask {
...
         ByteArrayOutputStream stderr = new ByteArrayOutputStream()
-        logger.debug("PackLibrariesTask - projectExec - $name:")
 
         try {
             Utils.projectExec(project, stdout, stderr, null, {
                 executable 'xcrun'
 
-                args 'lipo'
-                args '-create', '-output', "${outputLibDir}/lib${project.name}-j2objc.a"
+                args 'lipo', '-create'
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTask.groovy]: please resplit these.

when xcrun the first arg is basically the executable and it is nice to see that on a separate line.

in fact i would do:
```
executable 'xcrun'
args 'lipo'
(new line)
args '-create'
...
```

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTaskTest.groovy [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        // Expected Activity
+        MockProjectExec mockProjectExec = new MockProjectExec(proj, '/J2OBJC_HOME')
+        mockProjectExec.demandDeleteAndReturn(
+                "$proj.projectDir/build/packedBinaries/$proj.name-j2objcStaticLibrary/iosDebug")
+        mockProjectExec.demandExecAndReturn(
+                [
+                        'xcrun',
+                        'lipo', '-create',
+                        '-output', "/PROJECT_DIR/build/packedBinaries/$proj.name-j2objcStaticLibrary/iosDebug/lib$proj.name-j2objc.a",
+                        "/PROJECT_DIR/build/binaries/$proj.name-j2objcStaticLibrary/ios_arm64Debug/lib$proj.name-j2objc.a",
```
Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTaskTest.groovy]: do you want to get the list of archs from NativeCompilation.SUPPORTED_ARCHS instead?
you aren't testing that these are in fact the archs but are instead testing that the archs as used by NativeCompilation are matched by this task

**Comment by advayDev1** [comment]
lgtm with minor comments

**Comment by confile** [comment]
LGTM

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTask.groovy (hunk 1 above)
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTask.groovy]: Fixed in unit test as well

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTaskTest.groovy (hunk 2 above)
Comment (brunobowden) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTaskTest.groovy]: Update arguments to use ALL_SUPPORTED_ARCHS, e.g.:

    "/PROJECT_DIR/build/binaries/$proj.name-j2objcStaticLibrary/${NativeCompilation.ALL_SUPPORTED_ARCHS[2]}Debug/lib$proj.name-j2objc.a",

Then asserted on the size of the array:

        assert NativeCompilation.ALL_SUPPORTED_ARCHS.size() == 5, 'Need to update list of arguments above'

Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/PackLibrariesTaskTest.groovy]: FYI: an alternative would be
```
NativeCompilaton.ALL_SUPPORTED_ARCS.collect({ String arch ->
"/PROJECT_DIR/build/binaries/$proj.name-j2objcStaticLibrary/$arch}Debug/lib$proj.name-j2objc.a" })
```

**Pull request merged**